devito/types/sparse.py

```diff
@@ -120,6 +140,22 @@ def __subfunc_setup__(self, key, suffix, dtype=None):
         if key is not None and not isinstance(key, SubFunction):
             key = np.array(key)
 
+        # Check if already a SubFunction
+        d = self.indices[self._sparse_position]
```

Comment:
Move definition of `d` inside the `if isinstance` block? It only gets used there right?

Reply:
done